WordPress/build.gradle

```diff
@@ -716,3 +716,23 @@ if (project.hasProperty("debugStoreFile")) {
         }
     }
 }
+
+// Copy React Native JavaScript bundle and source map so they can be upload it to the Crash logging
+// service during the build process.
+android {
+    applicationVariants.all { variant ->
+        variant.mergeAssetsProvider.configure {
+            doLast {
+                // Copy bundle and source map files
+                copy {
+                    from(outputDir)
+                    into("${buildDir}/react-native-bundle-source-map")
+                    include("*.bundle", "*.bundle.map")
+                }
+
+                // Delete source maps
+                delete(fileTree(dir: outputDir, includes: ['**/*.bundle.map']))
+            }
+        }
+    }
+}
```

Comment:
I guess there was some transitive dependency bump, that's causing this problem in #20483 .

When I run `assembleJetpackJalapenoDebug` in current `trunk`, files are generated. When I do the same in #20483 , they're not.

Reply:
> I think this block of code wasn't executed. @fluiddot have you seen similar issue, or maybe you have an idea why it happens?
> 
> I guess there was some transitive dependency bump, that's causing this problem in #20483 .
> 
> When I run `assembleJetpackJalapenoDebug` in current `trunk`, files are generated. When I do the same in #20483 , they're not.

Hey @wzieba 👋 ! That's interesting. I haven't encountered this issue yet, and after checking other PRs, seems specific to the PR https://github.com/wordpress-mobile/WordPress-Android/pull/20483 as you pointed out.

This code block should be executed in every variant, so I wonder what's preventing it in that PR. One possibility is that the files aren't generated and so they are not copied. I can take a quick look in case I narrow down the culprit.

Reply:
As far as I've checked, the difference between `trunk` and https://github.com/wordpress-mobile/WordPress-Android/pull/20483 is this code block is only executed the first time. Subsequent builds don't trigger it. I only managed to trigger it again by modifying the `build.gradle` file (e.g. adding an empty line). This makes me think that there might be some kind of cache layer producing the issue. In fact, I compared the Buildkite logs between https://github.com/wordpress-mobile/WordPress-Android/pull/20483 and another one that succeeded and noticed that the task `:WordPress:mergeWordpressJalapenoDebugAssets` is marked as `FROM-CACHE` in the PR ([reference](https://buildkite.com/automattic/wordpress-android/builds/17663#018e4271-f47b-4ba1-a4a8-5a68c371292d/228-416)), whereas in other PR is not cached ([reference](https://buildkite.com/automattic/wordpress-android/builds/17691#018e51b2-f774-469f-b900-5ec1af145a8d/228-415)).

As a workaround, we could move the code block to a new Gradle task and ensure is not cached.

Reply:
Solving the convo as this will be fixed via https://github.com/wordpress-mobile/WordPress-Android/pull/20498.

Reply:
Sorry @fluiddot , I only now spotted your above comments. Thanks for responding here 🙇 , and yes, we continue the discussion in #20498 .